network-config-analyzer/NetworkConfigQueryRunner.py

```diff
@@ -15,6 +16,43 @@ def __init__(self, key_name, configs_array, expected_output, output_configuratio
         self.network_configs = network_configs
         self.expected_output_file = expected_output
 
+    def _parse_network_config_of_specified_policy(self, config_name):
+        """
+        User wants a specific policy from the given config.
+        config_name has one of the following forms:
+        (1) <config>/<namespace>/<policy>
+        (2) <config>/<layer>/<namespace>/<policy>/ , where layer is the relevant kind from
+                                                    ['k8s', 'calico', 'istio', 'ingress']
+        :param str config_name: the full config name (from which a specific policy is requested)
+        :return: the parsed config name, policy name, policy type
+        :rtype: (str, str, NetworkConfig.ConfigType)
+        """
+        sep_count = config_name.count('/')
+        split_config = config_name.split('/', 1)
+        config_name = split_config[0]
+        policy_type = None
+        if sep_count <= 2:
+            policy_name = split_config[1]
+        elif sep_count == 3:
+            split_layer = split_config[1].split('/', 1)
+            layer = split_layer[0]
+            policy_name = split_layer[1]
+            if layer not in ['k8s', 'calico', 'istio', 'ingress']:
+                raise Exception(f'Layer {layer} is not supported')
+            if layer == 'k8s':
+                policy_type = NetworkConfig.ConfigType.K8s
+            elif layer == 'calico':
+                policy_type = NetworkConfig.ConfigType.Calico
+            elif layer == 'istio':
+                policy_type = NetworkConfig.ConfigType.Istio
+            elif layer == 'ingress':
+                policy_type = NetworkConfig.ConfigType.Ingress
```

Comment:
Maybe the `NetworkConfig.ConfigType` class can provide this backwards translation service

Reply:
done